### PR TITLE
Clean up dependency on proxygen libs

### PR DESCRIPTION
**Pull request by dangleptr**
I found the problem when i write a UT with no dependency of proxygen lib.  
It is introduced by #210 and #395  

Please pay more attention to your dependency.  @steppenwolfyuetong @darionyaphet

### CODE REVIEW

**Comment by dangleptr** [comment]
Jenkins go

File: src/storage/test/StorageHttpHandlerTest.cpp [hunk 1]
```diff
@@ -43,32 +43,6 @@ class StorageHttpHandlerTestEnv : public ::testing::Environment {
...
-    auto mClient
-        = std::make_unique<meta::MetaClient>(threadPool, std::move(addrsRet.value()), true);
-    mClient->init();
-    std::string dataPath = folly::stringPrintf("%s/data", rootPath.path());
-    auto sc = TestUtils::mockStorageServer(mClient.get(),
-                                           dataPath.c_str(),
-                                           localIp,
-                                           localDataPort);
-
     {
```
Comment (steppenwolfyuetong) [inline src/storage/test/StorageHttpHandlerTest.cpp]: Can explain that the mockserver is not started, how can the latter test run?

File: src/graph/CMakeLists.txt [hunk 2]
```diff
@@ -47,5 +46,9 @@ add_dependencies(
     stats_obj
 )
 
+add_library(
+    graph_http_obj OBJECT
+    GraphHttpHandler.cpp
+)
```
Comment (steppenwolfyuetong) [inline src/graph/CMakeLists.txt]: Is there no dependency here?

Comment (dangleptr) [reply src/graph/CMakeLists.txt]: Just depend on base_obj.  Good catch

File: src/storage/test/StorageHttpHandlerTest.cpp (hunk 1 above)
Comment (dangleptr) [reply src/storage/test/StorageHttpHandlerTest.cpp]: We just need the http server.

File: src/graph/CMakeLists.txt (hunk 2 above)
Comment (dutor) [reply src/graph/CMakeLists.txt]: Here is adding a library, OBJECT library, no dependencies.

File: src/meta/CMakeLists.txt [hunk 3]
```diff
@@ -49,6 +48,13 @@ add_dependencies(
     stats_obj
 )
 
+add_library(
+    meta_http_handler OBJECT
+    MetaHttpHandler.cpp
+)
+add_dependencies(
+    meta_http_handler
+    base_obj)
```
Comment (steppenwolfyuetong) [inline src/meta/CMakeLists.txt]: right bracket format

**Comment by dutor** [comment]
Executables and shared libraries depend on other libraries, while static or OBJECT libraries depend on other headers, not libraries.

Now `add_dependencies` brings us too many chaos, we have to address this recently.

Please refer to this issue #406 .

**Comment by dutor** [comment]
Besides, by move `third-party` out from our repo, `add_dependencies` is no longer needed anymore.

**Comment by dangleptr** [comment]
> Executables and shared libraries depend on other libraries, while static or OBJECT libraries depend on other headers, not libraries.
> 
> Now `add_dependencies` brings us too many chaos, we have to address this recently.
> 
> Please refer to this issue #406 .

Make sense.  OBJECT should just depend on the headers.  
"add_dependencies" only work for Executables

**Comment by dangleptr** [comment]
> Besides, by moving `third-party` out from our repo, `add_dependencies` is no longer needed anymore.

Very good.

**Pull request merged**